src/Model/Resolver/Menu.php

```diff
@@ -84,43 +75,49 @@ public function resolve(
         ResolveInfo $info,
         array $value = null,
         array $args = null
-    ): Value {
-        $result = function () {
-            return null;
-        };
-
-        if (isset($args['identifier'])) {
-            $menu = $this->menuCollectionFactory->create();
-
-            /** Updated with identifier filtering */
-            $menu->addFieldToFilter('identifier', $args['identifier'])->load();
+    )
+    {
+        $identifier = $args['identifier'];
 
-            $menuData = $menu->getFirstItem()->getData();
+        $menu = $this->menuFactory
+            ->create()
+            ->load($identifier);
 
-            $items = $this->itemCollectionFactory->create();
-
-            /** Updated with menu id taken from menu data */
-            $items->addMenuFilter($menuData['menu_id'])
-                ->addStatusFilter()
-                ->setParentIdOrder()
-                ->setPositionOrder();
-
-            $menuData['items'] = $items->getData();
+        if ($menu->getId() === null) {
+            throw new \InvalidArgumentException("Could not find menu with identifier '${identifier}'");
+        }
 
-            foreach ($menuData['items'] as &$item) {
-                if (isset($item['category_id'])) {
-                    $category = $this->categoryRepository->get($item['category_id']);
-                    $item['url'] = sprintf('/%s', $category->getUrlPath());
-                }
-            }
+        return array_merge(
+            $menu->getData(),
+            [
+                'items' => $this->getMenuItems($menu['menu_id'])
+            ]);
+    }
 
-            if ($menuData) {
-                $result = function () use ($menuData) {
-                    return $menuData;
-                };
+    /**
+     * @param string $menuId
+     * @return array
+     */
+    private function getMenuItems(string $menuId): array
+    {
+        $menuItems = $this->itemCollectionFactory
+            ->create()
+            ->addMenuFilter($menuId)
+            ->addStatusFilter()
+            ->setParentIdOrder()
+            ->setPositionOrder()
+            ->getData();
+
+        return array_map(function ($item) {
+            if (isset($item[static::CATEGORY_ID_KEY])) {
+                $categoryUrlPath = $this->categoryRepository
+                    ->get($item[static::CATEGORY_ID_KEY])
```

Comment:
static is used for late static binding that is not needed here. Use self:: instead.
Here are the docs on late static binding: https://www.php.net/manual/en/language.oop5.late-static-bindings.php